projects/plugins/crm/includes/ZeroBSCRM.AJAX.php

```diff
@@ -836,6 +837,18 @@ function ZeroBSCRM_get_quote_template() {
 							$v = '';
 							if ( isset( $_POST['quote_fields'][ 'zbscq_' . $key ] ) ) {
 								$v = sanitize_text_field( $_POST['quote_fields'][ 'zbscq_' . $key ] );
+
+								// Here is where we search and replace placeholders for dates with a date string and date time strings), initially checking the value is similar to that of a Unix timestamp
+								if ( preg_match( '/^[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1])$/', $v ) ) {
```

Comment:
> initially checking the value is similar to that of a Unix timestamp

This doesn't check for a Unix timestamp, but instead a `YYYY-MM-DD` (ISO standard).

Reply:
Also, I'd avoid doing any "smart" processing on date-like content unless the field is really a date type.

Reply:
I might be better off changing the comment here as I realize it isn't accurate - the date will only ever be in the format `yyyy-mm-dd` when coming from the $_POST data due to how the date input sends date data via POST. So we can still check for that format to prevent extra processing (the regex is still correct in this respect, just the comment is not) - or just check if `is_numeric` and run the datetime function check on that.

> Also, I'd avoid doing any "smart" processing on date-like content unless the field is really a date type.

Not too sure what you mean here?

Reply:
> Not too sure what you mean here?

Just because something looks like a date field doesn't mean it is a date field; the best way to ensure it's a date field is to look at the object definition (each field has a specific type associated). For example, one could have a math problem of `1500-12-30`. Or someone could have an input of 31 February. But highly unlikely edge cases here, so probably safe enough to ignore.

Reply:
Following up here for completion as we discussed this in Slack - the $_POST data doesn't have any object related fields associated with keys, only the keys and their values.